.github/workflows/test.yml

```diff
@@ -4,31 +4,49 @@ on: [push, pull_request]
 
 jobs:
   run-tests:
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
     strategy:
       fail-fast: false
       matrix:
+        os: [windows-latest, macos-latest, ubuntu-latest]
         python-version: ["3.7", "3.8", "3.9", "3.10"]
     steps:
     - uses: actions/checkout@v2
     - name: Set up Python ${{ matrix.python-version }}
       uses: actions/setup-python@v2
       with:
         python-version: ${{ matrix.python-version }}
-    - name: Build packages, and install the wheel
+    - name: Install packages
+      run: |
+        pip install -U pip
+        pip install .
+    - name: Check Flake8
+      run: |
+        flake8 --version
+    - name: Check Black
+      run: |
+        black --version
+    - name: Run tests
+      run: |
+        cd tests
+        if [ "$RUNNER_OS" == "Windows" ]; then
+          WIN=1 bash ./run_tests.sh
+        else
+          bash ./run_tests.sh
+        fi
+      shell: bash
+    - name: Build package
       run: |
         python setup.py sdist --formats=gztar
         python -m pip install wheel
         python setup.py bdist_wheel
-        cd dist
-        python -m pip install flake8_black-*.whl
-        cd ..
-        # Wheel should now be installed
-        flake8 --version
-    - name: Run tests
+        ls dist
+    - name: Check if Manifest was complete
```

Comment:
The way we check the manifest was complete enough to run the tests is to unpack it and run the tests on the unpacked files. That functionality has been lost here (perhaps it was not explicit enough in the comments?)